#goal definition
string obj_link
bool ideal_attach

---
#result definition
string result

---
#feedback
string feedback
